### PR TITLE
Add parser for Elixir `mix.exs` files

### PR DESCRIPTION
**Pull request by timgluz**
Hi,

this PR adds parser for `mix.exs` files;

I followed those docs for implementation:

* [Introduction to Mix](https://elixir-lang.org/getting-started/mix-otp/introduction-to-mix.html)
* [Mix Deps module](https://hexdocs.pm/mix/Mix.Tasks.Deps.html#content)
* [Mix Version module](https://hexdocs.pm/elixir/Version.html)

**NB!** There's also `mix.lock` file, which is not implemented with this PR.

### CODE REVIEW

File: lib/versioneye/parsers/mix_parser.rb [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    content = content.to_s.gsub(/\s+/, ' ')
+
+    content.to_s.strip
+  end
+
+  def init_project(url = nil)
+    Project.new(
+      project_type: Project::A_TYPE_MIX,
+      language:  Product::A_LANGUAGE_ELIXIR,
+      url: url
```
Comment (reiz) [inline lib/versioneye/parsers/mix_parser.rb]: @timgluz What is this? The URL to the file? Or the URL to the project homepage?

**Comment by reiz** [comment]
@timgluz Looks good so far. Just wondering about the `project.url`.

File: lib/versioneye/parsers/mix_parser.rb (hunk 1 above)
Comment (reiz) [inline lib/versioneye/parsers/mix_parser.rb]: @timgluz Just noticing that url is always nil. In that case we can remove it. Right?

Comment (timgluz) [inline lib/versioneye/parsers/mix_parser.rb]: I dont know - i took that parameter from `GemfileParser` as i wanted to keep similar interface with other parsers. https://github.com/versioneye/versioneye-core/blob/master/lib/versioneye/parsers/gemfile_parser.rb#L262

Comment (reiz) [inline lib/versioneye/parsers/mix_parser.rb]: ok

**Pull request merged**